oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcClientRegistrationEndpointFilter.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.core.OAuth2ErrorCodes;
+import org.springframework.security.oauth2.core.OAuth2TokenType;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationResponseType;
+import org.springframework.security.oauth2.core.http.converter.OAuth2ErrorHttpMessageConverter;
+import org.springframework.security.oauth2.core.oidc.OidcClientRegistration;
+import org.springframework.security.oauth2.core.oidc.http.converter.OidcClientRegistrationHttpMessageConverter;
+import org.springframework.security.oauth2.server.authorization.OAuth2Authorization;
+import org.springframework.security.oauth2.server.authorization.OAuth2AuthorizationService;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClientRepository;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author Ovidiu Popa
+ * @since 0.1.1
+ */
+public class OidcClientRegistrationEndpointFilter extends OncePerRequestFilter {
+
+	private static final String SCOPE_CLAIM_DELIMITER = " ";
+
+	private final OidcClientRegistrationHttpMessageConverter clientRegistrationHttpMessageConverter =
+			new OidcClientRegistrationHttpMessageConverter();
+	private final RegisteredClientRepository registeredClientRepository;
+	private final OidcClientRegistrationToRegisteredClientConverter oidcClientToRegisteredClientConverter =
+			new OidcClientRegistrationToRegisteredClientConverter();
+	private final RegisteredClientToOidcClientRegistrationConverter registeredClientToOidcClientConverter =
+			new RegisteredClientToOidcClientRegistrationConverter();
+	private final HttpMessageConverter<OAuth2Error> errorHttpResponseConverter =
+			new OAuth2ErrorHttpMessageConverter();
+	private final RequestMatcher requestMatcher;
+	private final OAuth2AuthorizationService authorizationService;
+
+	public OidcClientRegistrationEndpointFilter(RegisteredClientRepository registeredClientRepository,
```

Comment:
Provider constructor that doesn't provide `oidcClientRegistrationUri` and uses default `DEFAULT_OIDC_CLIENT_REGISTRATION_ENDPOINT_URI` - see `OAuth2TokenEndpointFilter`